src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1786,6 +1789,9 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
                     if (msbuildProject != null)
                     {
                         msbuildProject.MSBuildNuGetProjectSystem.EndProcessing();
+
+                        // raise nuget batch end event
+                        msbuildProject.NotifyEndBatchProcessing();
```

Comment:
It doesn't look like either of these end calls actually check that the start call was made. Potentially listeners could get an end call when no batch started if a download task or anything above the start calls in the try fails.